src/criteria_assessment/query_thresholds.jl

```diff
@@ -298,6 +342,30 @@ function threshold_mask(
     return rebuild(view_of_data, sparse(convert.(UInt8, view_of_data)))
 end
 
+"""
+Handles threshold masking using the integrated assessment parameter struct
+"""
+function threshold_mask(
+    params::RegionalAssessmentParameters
+)::Raster
+    # build out a set of criteria filters using the regional criteria
+    # NOTE this will only filter over available criteria
+    filters = build_criteria_bounds_from_regional_criteria(params.regional_criteria)
+
+    # map our regional criteria 
+    @debug "Applying criteria thresholds to generate mask layer"
+    mask_layer = apply_criteria_thresholds(
+        # This is the raster stack
+        params.region_data.raster_stack,
+        # The slope table dataframe
+        params.region_data.slope_table,
+        # The list of criteria bounds
+        filters
+    )
```

Comment:
In these cases I would use inline comments:

```suggestion
    mask_layer = apply_criteria_thresholds(
        params.region_data.raster_stack,  # This is the raster stack
        params.region_data.slope_table,  # The slope table dataframe
        filters  # The list of criteria bounds
    )
```

Reply:
Disagree - blows out your line width! 

I won't change this unless you threaten me...